gw_eccentricity/eccDefinition.py

```diff
@@ -1350,26 +1442,85 @@ def measure_ecc(self, tref_in=None, fref_in=None):
         # check if eccentricity is monotonic and convex
         self.check_monotonicity_and_convexity()
 
-        # If tref_in is a scalar, return a scalar
-        if tref_in_ndim == 0:
-            self.mean_anomaly = self.mean_anomaly[0]
-            self.eccentricity = self.eccentricity[0]
-            self.tref_out = self.tref_out[0]
-
-        if fref_in is not None and fref_in_ndim == 0:
-            self.fref_out = self.fref_out[0]
-
         if self.debug_plots:
             # make a plot for diagnostics
             fig, axes = self.make_diagnostic_plots()
             self.save_debug_fig(fig, f"gwecc_{self.method}_diagnostics.pdf")
             plt.close(fig)
-        return_dict = {"eccentricity": self.eccentricity,
-                       "mean_anomaly": self.mean_anomaly}
-        if fref_in is not None:
-            return_dict.update({"fref_out": self.fref_out})
+        # return measured eccentricity, mean anomaly and reference time or
+        # frequency where these are measured.
+        return self.make_return_dict_for_eccentricity_and_mean_anomaly()
+
+    def set_eccentricity_and_mean_anomaly_to_zero(self):
+        """Set eccentricity and mean_anomaly to zero."""
```

Comment:
I'm thinking the following:
What if we just return an array of zeros, the same size as tref_in/fref_in and be done with it?
If the user asks for a tref_in/fref_in that is outside t, it's not a big deal because zero ecc remains zero ecc for all times. That way, this function is very simple, and you don't have to make assumptions about instantaneous f_22 being a decent option.

Reply:
Okay, done!